python-libraries/data-platform-catalogue/data_platform_catalogue/client/datahub/datahub_client.py

```diff
@@ -151,41 +151,51 @@ def upsert_data_product(self, metadata: DataProductMetadata):
 
         data_product_urn = "urn:li:dataProduct:" + "".join(name.split())
 
-        if metadata.domain:
```

Comment:
This was originally here to prevent domain being assigned if it was an empty string. 

Agree that this isn't the place for that test. But we'll need some validation in the entities section below.

Reply:
Agreed, however Domain is mandatory so shouldn't be an empty value.